libraries/AP_Airspeed/AP_Airspeed_NMEA.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ *   NMEA Sensor driver for VHW ans MTW messages over Serial
+ *   https://gpsd.gitlab.io/gpsd/NMEA.html#_vhw_water_speed_and_heading
+ *   https://gpsd.gitlab.io/gpsd/NMEA.html#_mtw_mean_temperature_of_water
+ */
+
+#include <AP_Vehicle/AP_Vehicle.h>
+
+#if APM_BUILD_TYPE(APM_BUILD_Rover) || APM_BUILD_TYPE(APM_BUILD_ArduSub) 
+
+#include "AP_Airspeed.h"
+#include "AP_Airspeed_NMEA.h"
+
+extern const AP_HAL::HAL &hal;
+
+AP_Airspeed_NMEA::AP_Airspeed_NMEA(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+bool AP_Airspeed_NMEA::init()
+{
+    const AP_SerialManager& serial_manager = AP::serialmanager();
+
+    _uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_AirSpeed, 0);
+    if (_uart == nullptr) {
+        return false;
+    }
+
+    _uart->begin(serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_AirSpeed, 0));
+
+    // make sure this sensor cannot be used in the EKF
+    set_use(0);
+
+    // must set use zero offset to pass offset check for health
+    set_use_zero_offset();
+
+    return true;
+}
+
+// read the from the sensor
+bool AP_Airspeed_NMEA::get_airspeed(float &airspeed)
+{
+    if (_uart == nullptr) {
+        return false;
+    }
+
+    // read any available lines from the sensor
+    float sum = 0.0f;
+    uint16_t count = 0;
+    int16_t nbytes = _uart->available();
+    while (nbytes-- > 0) {
+        char c = _uart->read();
+        if (decode(c)) {
+            if (_sentence_type == TYPE_VHW) {
+                sum += _speed;
+                count++;
+            } else {
+                _temp_sum += _temp;
+                _temp_count++;
+            }
+        }
+    }
+
+    if (count == 0) {
+        // Cant return false because updates are too slow, return previous reading
+        // Could return false after some timeout, however testing shows that the DST800 just stops sending the message at zero speed
+        airspeed = _last_speed;
+    } else {
+        // return average of all measurements
+        airspeed = sum / count;
+        _last_speed = airspeed;
+    }
+
+    return true;
```

Comment:
check if received data in last 0.5s and unhealthy if not?

Reply:
Done, used a 2s threshold, the sensors are slow and I don't have it anymore to check what we should really use.